src/vs/editor/common/diff/standardLinesDiffComputer.ts

```diff
@@ -153,11 +153,18 @@ export class StandardLinesDiffComputer implements ILinesDiffComputer {
 		return new LinesDiff(changes, moves, hitTimeout);
 	}
 
+	private shouldDynamicProgrammingDiff(m: number, n: number) {
+		const [a, b] = m > n ? [n, m] : [m, n];
+
+		// Myers D should greater than (or equal to) `b - a`, when `D > a` we choose DP diff.
+		return b - a > a;
```

Comment:
```suggestion
		return b > 2*a;
```

I don't think this makes sense, as dynamic programming is always slower than Myers.

Reply:
I test with the following result, Myers costs much more than DP:
<img width="1392" alt="Screen Shot 2023-06-18 at 09 15 29" src="https://github.com/microsoft/vscode/assets/18526463/7a88a196-46d8-4d11-ae6a-98208a18b63d">
<img width="1440" alt="Screen Shot 2023-06-18 at 09 16 00" src="https://github.com/microsoft/vscode/assets/18526463/9b8ec38b-2427-4116-aeb5-e49174c90884">

I think it's because Myers is O(ND) and DP is O(MN), we should compare D and M to consider which is proper.

Reply:
But D is always smaller than M! Maybe something else is off in the Myers implementation. Will investigate.

Reply:
Maybe not, according to the [paper](https://neil.fraser.name/writing/diff/myers.pdf), D refers to "the size of the minimum edit script", which should satisfy N - M <= D <= N + M(subsequence, e.g. 'aa' to 'aabb', takes N - M, and no common character, e.g. 'abc' to 'xyz', takes N + M), even N - M is not definitely smaller than M.

I'm not sure if there is a way to handle the worst case, and let D be always smaller than M. Looking forward to your conclusion~